vendingMachine.js

```diff
@@ -1,68 +1,101 @@
+        //Calculate the spaces required on the left and right to organize the stock of the product in column 2.
+        spaceLeft2 = SPACES.repeat(Math.round((spaceTable - (("Stock:" + productStock[i][j - 1]).length)) / 2))
+        spaceRight2 = SPACES.repeat((productStock[i][j - 1].length % 2 != 1) ? spaceLeft2.length : spaceLeft2.length - imparsubtraction);
+        //Calculate the spaces required on the left and right to organize the stock of the product in column 2.
+        spaceLeft3 = SPACES.repeat(Math.round((spaceTable - (("Stock:" + productStock[i][j]).length)) / 2))
+        spaceRight3 = SPACES.repeat((productStock[i][j].length % 2 != 1) ? spaceLeft3.length : spaceLeft3.length - imparsubtraction);
+        //Organize the results in the coordinates and stock variables, to add it to the itemTable variable and print the final shade.
+        coord = `|${spaceLeft}${(i + 1)}.${(j - 1)}${spaceRight}|${spaceLeft}${(i + 1)}.${(j )}${spaceRight}|${spaceLeft}${(i + 1)}.${(j + 1)}${spaceRight}|`;
+        stock = `|${spaceLeft1}Stock:${matrixStock[i][j - 2]}${spaceRight1}|${spaceLeft2}Stock:${matrixStock[i][j - 1]}${spaceRight2}|${spaceLeft3}Stock:${matrixStock[i][j]}${spaceRight3}|`;
+        itemTable = itemTable + `|\n${stock}\n${coord}\n|${topLine}\n`;
       }
     }
   }
-  console.log(result);
+  console.log(itemTable);
 }
 
-function productStock(arr3, a, b) {
-  if (a <= arr3.length & b <= arr3.length) {
-    let newStock = arr3[a][b];
-    if (newStock !== 0) {
+function showStock(matrixStock, row, column) {
+  if (row <= matrixStock.length & column <= matrixStock.length) {
+    let newStock = matrixStock[row][column];
+    //When the product stock is different from 0, it shows the quantity by subtracting, otherwise it prints a message.
+    if (newStock != 0) {
       newStock = newStock - 1;
-      arr3[a][b] = newStock;
+      matrixStock[row][column] = String(newStock);
     } else {
-      console.log('El producto ya no se encutra disponible');
+      console.log('The product is no longer available');
     }
   }
 }
 
-function productRequest(matrix, arr3) {
-  let a, b, code;
+function requestProduct(matrixProducts, matrixStock) {
+  let rowPosition, columnPosition, codeProduct;
+  let showProduct = '';
   while (1) {
-    code = prompt('Ingerese el codigo numerico del producto: ');
-    if (code[1] === '.' & code[0] <= matrix.length & code[2] <= matrix.length) {
-      a = code[0] - 1, b = code[2] - 1;
-      let prod = '';
-      prod = matrix[a][b];
-      console.log("Disfrute su producto: " + prod);
-      productStock(arr3, a, b);
+    codeProduct = prompt('Enter the product code number: ');
+    //If it is a valid code, it prints on console the corresponding product, otherwise it prints an invalid code message.
+    if (codeProduct[1] === '.' & codeProduct[0] <= matrixProducts.length & codeProduct[2] <= matrixProducts.length) {

```

Comment:
apply the not logic instead.

`  if(!Good Format) return console.log('wrong code');`

Reply:
To solve it use the "return" statement, to avoid the use of "else", but by carrying out the "return" instruction, it causes the "while" loop to terminate its operation.